docs/src/advanced_topics/execution_trace_optimization.md

```diff
@@ -0,0 +1,44 @@
+# Execution trace optimization
+
+## Understanding cycle counts in Miden VM
+
+When we refer to "number of cycles" in most Miden VM documentation, we're specifically referring to the **stack rows** portion of the execution trace. However, the actual proving time is determined by what we call the "true number of cycles," which is the maximum of all trace segment lengths:
+
+- **Stack rows**: One row per VM operation (what `clk` outputs)
+- **Range checker rows**: Added when opcodes request 16-bit range checks
+- **Chiplet rows**: Added when opcodes call specialized chiplets
+
+The **true number of cycles** = `max(stack_rows, range_checker_rows, chiplet_rows)`
```

Comment:
I would distinguish here between "number of cycles" and "number of rows". Specifically, since "cycles" refers to the length of stack rows, here we're referring more to the "number of rows in the final trace".

I would also mention (wherever it makes most sense):
- nit: whatever the `max()` is, it gets rounded up to the next power of 2
- the other 2 sets of columns get padded up to the max (well, power of 2)
  - note that "padding" means "whatever values make the constraints work", but can be intuitively thought of as "setting the cells to 0"